src/schema/users.ts

```diff
@@ -658,50 +664,28 @@ export const resolvers: IResolvers<any, Context> = {
       const user = await repo.findOne({ id: ctx.userId });
 
       if (!user) {
-        throw new AuthenticationError(
-          JSON.stringify({ auth: 'User not found' }),
-        );
-      }
-
-      const isUsernameValid = new RegExp(/^@?(\w){1,15}$/).test(data.username);
-      const isGithubValid = new RegExp(/^@?([\w-]){1,39}$/i).test(data.github);
-      const isTwitterValid = new RegExp(/^@?(\w){1,15}$/).test(data.twitter);
-      const isHashnodeValid = new RegExp(/^@?([\w-]){1,39}$/i).test(
-        data.hashnode,
-      );
-
-      if (data.username && !isUsernameValid) {
-        throw new ValidationError(
-          JSON.stringify({ username: 'username is invalid' }),
-        );
+        throw new AuthenticationError('Unauthorized!');
       }
 
-      if (data.github && !isGithubValid) {
-        throw new ValidationError(
-          JSON.stringify({ github: 'github is invalid' }),
-        );
-      }
-
-      if (data.twitter && !isTwitterValid) {
-        throw new ValidationError(
-          JSON.stringify({ twitter: 'twitter is invalid' }),
-        );
-      }
-
-      if (data.hashnode && !isHashnodeValid) {
-        throw new ValidationError(
-          JSON.stringify({ hashnode: 'hashnode is invalid' }),
-        );
+      const regexParams: ValidateRegex[] = [
+        ['username', data.username, new RegExp(/^@?(\w){1,39}$/)],
+        ['github', data.github, new RegExp(/^@?([\w-]){1,39}$/i)],
+        ['twitter', data.twitter, new RegExp(/^@?(\w){1,15}$/)],
+        ['hashnode', data.hashnode, new RegExp(/^@?([\w-]){1,39}$/i)],
+      ];
+      const regexResult = validateRegex(regexParams);
+      if (Object.keys(regexResult).length) {
+        throw new ValidationError(JSON.stringify(regexResult));
```

Comment:
We just have to make sure the client knows how to parse this error and present it